### ISSUE TITLE
Undefined method error when a VEVENT component includes a DESCRIPTION with some IRIs

### ISSUE THREAD

**Issue by fabacab** [Rakefire/jekyll-ical-tag#25]
Hi again. I noticed a new issue with certain iCalendar feeds that I'm having difficulty troubleshooting because of my lack of Ruby skill. The error is consistent and happens during a `jekyll build` for me:

```
Liquid Exception: undefined method `force_encoding' for #<Icalendar::Values::Array:0x00007fb0623e7528> in /_layouts/calendars.html
jekyll 3.9.0 | Error:  undefined method `force_encoding' for #<Icalendar::Values::Array:0x00007fb0623e7528>
```

I've attached a copy of the iCalendar file ([serenoregis.ical.txt](https://github.com/Rakefire/jekyll-ical-tag/files/5603590/serenoregis.ical.txt)) causing the issue. It was sourced from [an Italian Web site's event calendar feed](https://serenoregis.org/appuntamenti/?ical=1).

The issue seems to be on [line 69 of `event.rb`](https://github.com/Rakefire/jekyll-ical-tag/blob/master/lib/jekyll-ical-tag/event.rb#L69), where the `description` variable remains of type [`Icalendar::Values::Array`](https://www.rubydoc.info/gems/icalendar/2.7.0/Icalendar/Values/Array) despite its conversion to a `string`.

There are three `VEVENT` components in the attached copy of the feed, but only one of them (the last of the three) is causing the error. It's also worth noting that it's also the only `VEVENT` component with a URL in the `DESCRIPTION` property, triggering the regex to match URLs and thus the relevant `description_urls` method.

Despite tracing this down in this way, I'm unsure what a good resolution would be so I'm opening this issue to ask for your advice on how to patch.

**Comment by rickychilcott** [comment]
Thanks for the documentation and hunting this down. I’m refactoring a few things and finally building out a test suite to test against this and other feeds. I hope to have a better understanding of how to fix after that. Give me a day or two to get back to you.

**Comment by fabacab** [comment]
> Give me a day or two to get back to you.

No rush. Let me know if there's any additional information I can offer.

**Comment by rickychilcott** [comment]
I think I fixed #25 by first checking if descriptions are an array, and if so, joining them with a new line character.

The exception is no longer raised @fabacab. Do you mind trying the latest (v1.2.5) to see if you're able to parse it correctly and you have the expected output?

Thanks for promoting this. I finally have a test suite as introduced in https://github.com/Rakefire/jekyll-ical-tag/commit/abbd080cc2a668b69f68332ad2f75340684e5fef This should make further regression testing (and refactoring) much easier.

**Comment by fabacab** [comment]
Thanks for getting around to this so quickly, @rickychilcott. While I don't get an error after updating the Gem to release `v1.2.5`, I'm not seeing the expected output. These are probably regressions so I'll record them individually as new issues for you momentarily.
